jerry-core/config.h

```diff
@@ -107,4 +189,128 @@
  */
 #define CONFIG_ECMA_GC_NEW_OBJECTS_SHARE_TO_START_GC (16)
 
+
+/**
+ * Sanity check for macros to see if the values are 0 or 1
```

Comment:
Are these sanity checks are generated somehow, or they should be written by hand? If the second, it would be good to add a note for add these checks whenever a new feature guard is introduced.

Reply:
Also, as the feature guards are grouped by categories (es5.1, es2015, other), I think these checks should be separated as well for the easier maintenance. (Probably a comment is enough)

Reply:
I'll add comments for these.